### ISSUE TITLE
Fix S3220 FP: Rule does not take into account access modifier of the members

### ISSUE THREAD

**Issue by csaba-sagi-sonarsource** [SonarSource/sonar-dotnet#5430]
### Description

Rule S3220 raises an issue even though the method that is overloaded is not accessible at the invocation.
The issue was reported by community: https://community.sonarsource.com/t/false-positive-s3220-method-calls-should-not-resolve-ambiguously-to-overloads-with-params-checks-private-constructors/58681

### Repro steps
```

public class SomeException : Exception
  {
      public Guid? ErrorReference { get; private set; }

      private SomeException (Guid errorReference, string message)
          : base(message)
      {
          ErrorReference = errorReference;
      }

      public SomeException (Guid errorRef, string message, params object[] args)
          : this(errorRef, string.Format(message, args))
      {

      }
  }

public class Test
{
    public Test()
    {
        var x = new SomeException(Guid.NewGuid(), "msgWithoutParams");
    }
} 

```
### Expected behavior

No issue is reported.

### Actual behavior

An issue is raised.

### Known workarounds

None.

### Related information

* C#/VB.NET Plugins version: [8.36.0.43782](https://github.com/SonarSource/sonar-dotnet/releases/tag/8.36.0.43782)
